src/Chapter3.hs

```diff
@@ -971,6 +1087,32 @@ implement the following functions:
 🕯 HINT: to implement this task, derive some standard typeclasses
 -}
 
+data WeekDay
+  = Monday
+  | Tuesday
+  | Wednesday
+  | Thursday
+  | Friday
+  | Saturday
+  | Sunday
+  deriving (Eq, Ord, Enum, Bounded, Show)
+
+isWeekend :: WeekDay -> Bool
+isWeekend weekDay
+  | weekDay > Friday = True
+  | otherwise        = False
+
+nextDay :: WeekDay -> WeekDay
+nextDay = succ
+
+daysToParty :: WeekDay -> Int
+daysToParty = go 0
+  where
+    go :: Int -> WeekDay -> Int
+    go n Friday = n
+    go n Sunday = go (n + 1) (minBound)
+    go n weekDay = go (n + 1) (nextDay weekDay)
```

Comment:
Interesting helper function! Generally, you also can solve it with the usage of `Enum` instance and `mod .. 7` function 😉

Reply:
No clue here, sorry 😒

Reply:
```haskell
daysToParty :: Weekday -> Int
daysToParty wd = (fromEnum Fri - fromEnum wd) `mod` 7
```

🙂 Hope this helps understanding!

Reply:
Okay, was simple enough...